tests/python_package_test/test_engine.py

```diff
@@ -453,3 +453,17 @@ def test_pandas_categorical(self):
         np.testing.assert_almost_equal(pred0, pred2)
         np.testing.assert_almost_equal(pred0, pred3)
         np.testing.assert_almost_equal(pred0, pred4)
+
+    def test_subset_train_val(self):
+        '''
+        Tests that it's fine to construct a single lgb.Dataframe object,
+        takes subsets of it, and uses the subsets for training and validation
+        '''
+        n = 1000
+        X = np.random.normal(size=n * 2).reshape(n, 2)
+        y = np.random.normal(size=n)
+        tmp_dat = lgb.Dataset(X, y)
+        tmp_dat_train = tmp_dat.subset(np.arange(int(n * .8)))
+        tmp_dat_val = tmp_dat.subset(np.arange(int(n * .8), n))
```

Comment:
eh...you don't need to add another dataset, just make tmp_dat_val be subset of tmp_dat_train. You don't need to consider if the training make senses, we just want to test dataset constructing. You can simply use tmp_data_train.construct() to test rather than calling lgb.train().